### PR TITLE
old style definition warning fixes

### LINKED ISSUE #2679: warning elimination: old style definition
As the result of #1302 we want to eliminate some special warnings. For this reason I open issues with the different warning types to split the problem into subtasks.

The resolution would be a PR that fixes the mentioned warnings, and - to defend against reintroducing such warnings - these kind of warnings should be added in the top level Makefile.am, to the end of AM_CFLAGS in the ENABLE_EXTRA_WARNINGS block. As travis transforms such warnings to errors, we can be sure no new warnings will be added in the future from these types.

This issue should cover the old style definition warning. I am splitting issue from the ticket #1965

 2 [-Wold-style-definition]
(The numbers in the statistics are just an approximate)

Hint: extra warnings can be enabled with --enable-extra-warnings configure option.

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
resolves #2679

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by furiel** [comment]
@kira-syslogng ok to test

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: Makefile.am [hunk 1]
```diff
@@ -84,7 +84,8 @@ AM_CFLAGS += \
 	-Wpointer-arith \
 	-Wpointer-sign \
 	-Wmissing-format-attribute \
-	-Wsuggest-attribute=noreturn
+	-Wsuggest-attribute=noreturn \
```
Comment (Kokan) [inline Makefile.am]: Please add this to cmake as well.

Comment (mehul-m-prajapati) [reply Makefile.am]: @Kokan : Okay.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**